barretenberg/cpp/src/barretenberg/honk/proof_system/ultra_verifier.cpp

```diff
@@ -114,15 +114,14 @@ template <typename Flavor> bool UltraVerifier_<Flavor>::verify_proof(const plonk
     // Execute Sumcheck Verifier
     auto sumcheck = SumcheckVerifier<Flavor>(circuit_size);
 
-    std::optional sumcheck_output = sumcheck.verify(relation_parameters, transcript);
+    auto [multivariate_challenge, purported_evaluations, sumcheck_verified] =
+        sumcheck.verify(relation_parameters, transcript);
 
-    // If Sumcheck does not return an output, sumcheck verification has failed
-    if (!sumcheck_output.has_value()) {
+    // If Sumcheck did not verify, return false
+    if (!sumcheck_verified.value()) {
```

Comment:
I think you should also check `has_value()` here

Reply:
you're right, I was being lazy. Done